vespalib/src/vespa/vespalib/net/tls/impl/openssl_crypto_codec_impl.h

```diff
@@ -37,6 +37,11 @@ class OpenSslCryptoCodecImpl : public CryptoCodec {
     size_t min_encode_buffer_size() const noexcept override {
         return MaximumTlsFrameSize;
     }
+    size_t min_decode_buffer_size() const noexcept override {
+        // Technically this would be MaximumFramePayloadSize, but we like power
```

Comment:
I thought it was the other way around; 16k payload plus some overhead for the frame, which would make the 100 bytes able to be an upper bound estimate. If it is 16k total encrypted frame size minus overhead for payload, the 100 bytes need to be more than a guess... (unless we want partial frames)

Reply:
Correct (and good catch), my assumptions were based on a misunderstanding of what a TLS "record" actually entails. After poking at various TLS RFCs, it seems the max overhead is actually significantly larger:

https://tools.ietf.org/html/rfc8449:
> Receiving large protected records can be particularly difficult for a
> device with limited operating memory.  TLS versions 1.2 [RFC5246] and
> earlier permit senders to generate records 16384 octets in size, plus
> any expansion from compression and protection up to 2048 octets
> (though typically this expansion is only 16 octets).  TLS 1.3 reduces
> the allowance for expansion to 256 octets.  Allocating up to 18K of
> memory for ciphertext is beyond the capacity of some implementations.

I.e. `min_encode_buffer_size()` --> 16384 + 2048 (since we're on TLSv1.2), `min_decode_buffer_size()` --> 16384

Will update the code to reflect this.